File: go/PeakInterval/PeakInterval.go
package main

import "math"

// Time -> 0(N) Space -> O(1)
func peak(intervals []int) int {
	start := math.MinInt

	for idx := range intervals {
		currVal := intervals[idx]
		end := 0
		if idx+1 == len(intervals) {
			end = math.MinInt
		} else {
			end = intervals[idx+1]
		}

		if currVal > start && currVal > end {
			return idx
		}

		start = currVal
	}
	return -1
}

File: python/peakInterval.py
# Time -> 0(N) Space -> 0(1)
def peak(intervals: list[int]) -> int:
    start = float("-inf")
    for idx in range(len(intervals)):
        curr_val = intervals[idx]
        end = float("-inf") if idx + 1 == len(intervals) else intervals[idx + 1]

        if curr_val > start and curr_val > end:
            return idx

        start = curr_val
    return -1


print(peak([]))
